# circRNA_neuro
In this repository you will find  source code and data sets related to our manuscript:
## Circular RNAs arising from synaptic host genes are modulated by SFPQ RNA-binding protein and increased during human neuronal differentiation

Michelle Watts (1,2), Marika Oksanen (1,2,3), Sanna Lejerkrans (1,2), Francesca Mastropasqua (1,2), Myriam Gorospe (3) and Kristiina Tammimies (1,2)

1.  Center of Neurodevelopmental Disorders (KIND), Centre for Psychiatry Research, Department of Women’s and Children’s Health, Karolinska Institutet and Stockholm Health Care Services, Region Stockholm, Stockholm, Sweden.
2.  Astrid Lindgren Children’s Hospital, Karolinska University Hospital, Region Stockholm, Stockholm, Sweden. 
3.  Laboratory of Genetics and Genomics, National Institute on Aging Intramural Research Program, NIH, Baltimore, MD, USA.

Correspondence: [kristiina.tammimies@ki.se](mailto:kristiina.tammimies@ki.se)

**Currently available as a preprint at bioRxiv DOI: [10.1101/2022.10.05.510949](https://doi.org/10.1101/2022.10.05.510949)**



### This resource is divided into the following three subdirectories:

#### 1. Differential Expression and Exon Usage Analysis
Here you will find raw count files for circRNAs (reads spanning back-spliced junctions), linear RNAs, and exons.
The R markdown file includes all codes used for differential expression/exon usage analysis as well as gene ontology and correlative analyses.

#### 2. Viariant Analysis
This directory contains code and data files used to analyse variants in circRNA-forming exons.

#### 3. Intron Analysis
Code and data required for analysis of circRNA flanking introns can be found in this directory.
